packages/jest-emotion/test/react-enzyme.test.js

```diff
@@ -98,3 +111,20 @@ test('enzyme test with prop containing css element with other label', () => {
   )
   expect(tree).toMatchSnapshot()
 })
+
+test('enzyme test with prop containing css element in fragment', () => {
+  useDeepSerializer = true
```

Comment:
couldn't matcher be extended differently (or used differently?) to avoid this beforeEach etc dance?

Reply:
The trouble is that matchers are global in the test file and there is no API for them to be removed.

If it was added within this test there is a danger that someone will add another test after and not intend for it to use the deep serializer.

If tests need to use both modes, they have to be switched by modifying the test function and that requires a before/afterEach to reset the mode between tests.

If the existing tests can be converted to using deep mode, we don't have the problem. I just didn't want to touch existing tests.

Reply:
cant a custom matcher use snapshots API? we could maybe just extend expect with separate `.toMatchDeepSnapshot` or smth like this, not sure if this is possible though

Reply:
Doesn't look any easier.

Reply:
Wouldn't using such approach - https://jestjs.io/docs/en/expect#custom-snapshot-matchers - be a cleaner solution?

Reply:
@Andarist does it mean someone should have knowledge of what's a deep snapshot? I honestly had no clue of why my tests were failing, I just did expect them to work.

Reply:
It certainly shouldn't crash. I'm not using enzyme, so I might be missing something here, but... I'm not talking here about adding some extra public API - which mode of the serializer you are using in your app is a separate thing from how our tests here are structured. And for the latter, we could just extend jest with 2 different matchers to have clean APIs for testing both modes of that enzyme serializer used in combination with emotion.

Reply:
The problem is serialization, not matchers. Serialization relies on globally installed serializers, regardless of how they are used in matchers.

If we don't want to touch the global serializer, we cannot test using `expect().toMatchSnapshot()`. Instead we need to figure out the internal jest serializer API, create a local serializer and test with something like `expect(localSerialize(wrapper)).toBe('x')`.

It only needs to be added to a custom matcher if it needs to be reused.

Reply:
I've create a proof of concept of what I had in mind - you can check it out here: https://github.com/liamcmitchell-sc/emotion/pull/1